packages/cashier/src/Stores/__tests__/general-store.spec.js

```diff
@@ -0,0 +1,520 @@
+import { waitFor } from '@testing-library/react';
+import { routes } from '@deriv/shared';
+import GeneralStore from '../general-store';
+import CashierNotifications from 'Containers/cashier-notifications.jsx';
+
+let cashier_menu, general_store, root_store, WS;
+
+beforeEach(() => {
+    root_store = {
+        client: {
+            account_list: [{ is_virtual: 0, title: 'USD' }],
+            account_status: {
+                status: [],
+                cashier_validation: [],
+            },
+            balance: null,
+            currency: 'USD',
+            has_active_real_account: false,
+            is_logged_in: true,
+            is_eu: false,
+            is_virtual: false,
+            switched: false,
+        },
+        common: {
+            routeTo: jest.fn(),
+        },
+        menu: {
+            attach: jest.fn(),
+            update: jest.fn(),
+        },
+        modules: {
+            cashier: {
+                account_prompt_dialog: {
+                    last_location: null,
+                    resetIsConfirmed: jest.fn(),
+                },
+                account_transfer: {
+                    accounts_list: [],
+                    container: 'account_transfer',
+                    selected_from: { balance: null },
+                    sortAccountsTransfer: jest.fn(),
+                },
+                crypto_fiat_converter: {
+                    converter_from_amount: null,
+                },
+                iframe: {
+                    clearIframe: jest.fn(),
+                },
+                onramp: {
+                    is_onramp_tab_visible: false,
+                },
+                payment_agent: {
+                    verification: {
+                        clearVerification: jest.fn(),
+                    },
+                    getAllPaymentAgentList: jest.fn().mockResolvedValue(['PA1', 'PA2']),
+                    setAllPaymentAgentList: jest.fn(),
+                    setPaymentAgentList: jest.fn().mockResolvedValueOnce([]),
+                },
+                payment_agent_transfer: {
+                    is_payment_agent: false,
+                    checkIsPaymentAgent: jest.fn(),
+                },
+                transaction_history: {
+                    is_crypto_transactions_visible: false,
+                    onMount: jest.fn(),
+                    setIsCryptoTransactionsVisible: jest.fn(),
+                },
+                withdraw: {
+                    check10kLimit: jest.fn().mockResolvedValueOnce(),
+                    setIsWithdrawConfirmed: jest.fn(),
+                    verification: {
+                        clearVerification: jest.fn(),
+                    },
+                },
+            },
+        },
+        ui: {
+            toggleSetCurrencyModal: jest.fn(),
+        },
+    };
+    WS = {
+        authorized: {
+            p2pAdvertiserInfo: jest.fn().mockResolvedValueOnce({ error: { code: 'advetiser_error' } }),
```

Comment:
Please rename `advetiser` to `advertiser` in all places.

Reply:
@farrah-deriv sure, thanks for the notice